tools/ci_build/rewriter.py

```diff
@@ -0,0 +1,221 @@
+
+        return True #end of fill_version_map(...)
+
+
+    def need_comment(op_type, opset_from, opset_to, domain):
+        '''callback func to check if the op is in ops'''
+
+        if op_type in operators and domain in operators[op_type]:
+            opset_to_index = version_map[op_type][domain].index(opset_to)
+
+            if opset_to_index == len(version_map[op_type][domain]) - 1:
+                opset_to = 9999 #if is the latest, extend to unlimited
+
+            elif opset_from == opset_to:
+                opset_to = version_map[op_type][domain][opset_to_index + 1]
+
+            for opset in range(opset_from, opset_to + 1):
+                if opset in operators[op_type][domain]:
+                    return False #do not comment
+
+        return True #end of need_comment(...)
+
+
+    def process_lines(lines, offset, end_mark, call_back):
+        '''extract op info from a logic code line start from offset to the line end
+           with end_mark, then trigger callback(op_type, opset_from, opset_to, domain)
+           return next line offset and whether current lines are disabled
+        '''
+
+        lines_to_process = []
+        while True: #collect the logical code line
+            lines_to_process.append(lines[offset])
+            stripped = lines[offset].strip()
+            if stripped.endswith(end_mark):
+                break
+            offset += 1
+        code_line = ''.join([line.strip() for line in lines_to_process])
+
+        disabled = False
+        if onnx_op in code_line:
+            trim_at = code_line.index(onnx_op) + onnx_op_len
+            args = [arg.strip() for arg in code_line[trim_at: -len(end_mark)].split(',')]
+            disabled = call_back(args[-1], int(args[-2]), int(args[-2]), args[-3])
+
+        elif onnx_typed_op in code_line:
+            trim_at = code_line.index(onnx_typed_op) + onnx_typed_op_len
+            args = [arg.strip() for arg in code_line[trim_at: -len(end_mark)].split(',')]
+            disabled = call_back(args[-1], int(args[-3]), int(args[-3]), args[-4])
+
+        elif onnx_versioned_op in code_line:
+            trim_at = code_line.index(onnx_versioned_op) + onnx_versioned_op_len
+            args = [arg.strip() for arg in code_line[trim_at: -len(end_mark)].split(',')]
+            disabled = call_back(args[-1], int(args[-3]), int(args[-2]), args[-4])
+
+        elif onnx_versioned_typed_op in code_line:
+            trim_at = code_line.index(onnx_versioned_typed_op) + onnx_versioned_typed_op_len
+            args = [arg.strip() for arg in code_line[trim_at: -len(end_mark)].split(',')]
+            disabled = call_back(args[-1], int(args[-4]), int(args[-3]), args[-5])
+
+        return offset + 1, disabled #end of process_lines(...)
+
+
+    lines = []
+    with open(file_path, 'r') as file_to_read:
+        lines = file_to_read.readlines()
+
+    shutil.move(file_path, file_path + '.bak')
+    with open(file_path, 'w') as file_to_write:
```

Comment:
What if you run build.py with reduced ops twice?
Will you lose the original cpu_execution_provider.cc, since the new .bak file will override the old bak file?

Reply:
Good point - fixing the line to avoid overwrite.